### ISSUE TITLE
Vulkan errors in 21_descriptor_layout

### ISSUE THREAD

**Issue by kurozenzen** [KyleMayes/vulkanalia#107]
Hey,

Thanks for the amazing tutorials. They explain really well what Vulkan does and how you use it.

Unfortunately, I have hit a bit of a roadblock in chapter 21. I am getting these errors when creating a pipeline.

```log
 ERROR 21_descriptor_layout > (VALIDATION) Validation Error: [ VUID-vkCmdDrawIndexed-None-02697 ] Object 0: handle = 0xe7e6d0000000000f, type = VK_OBJECT_TYPE_PIPELINE; Object 1: handle = 0x967dd1000000000e, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; Object 2: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x9888fef3 | vkCmdDrawIndexed(): VkPipeline 0xe7e6d0000000000f[] defined with VkPipelineLayout 0x967dd1000000000e[] is not compatible for maximum set statically used 0 with bound descriptor sets, last bound with VkPipelineLayout 0x0[] The Vulkan spec states: For each set n that is statically used by the VkPipeline bound to the pipeline bind point used by this command, a descriptor set must have been bound to n at the same pipeline bind point, with a VkPipelineLayout that is compatible for set n, with the VkPipelineLayout used to create the current VkPipeline, as described in Pipeline Layout Compatibility (https://vulkan.lunarg.com/doc/view/1.2.198.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdDrawIndexed-None-02697)
 ERROR 21_descriptor_layout > (VALIDATION) Validation Error: [ UNASSIGNED-CoreValidation-DrawState-DescriptorSetNotBound ] Object 0: handle = 0x2bf5b2473e0, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0xcde11083 | vkCmdDrawIndexed(): VkPipeline 0xe7e6d0000000000f[] uses set #0 but that set is not bound.
 ERROR 21_descriptor_layout > (VALIDATION) Validation Error: [ VUID-vkCmdDrawIndexed-None-02697 ] Object 0: handle = 0xe7e6d0000000000f, type = VK_OBJECT_TYPE_PIPELINE; Object 1: handle = 0x967dd1000000000e, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; Object 2: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x9888fef3 | vkCmdDrawIndexed(): VkPipeline 0xe7e6d0000000000f[] defined with VkPipelineLayout 0x967dd1000000000e[] is not compatible for maximum set statically used 0 with bound descriptor sets, last bound with VkPipelineLayout 0x0[] The Vulkan spec states: For each set n that is statically used by the VkPipeline bound to the pipeline bind point used by this command, a descriptor set must have been bound to n at the same pipeline bind point, with a VkPipelineLayout that is compatible for set n, with the VkPipelineLayout used to create the current VkPipeline, as described in Pipeline Layout Compatibility (https://vulkan.lunarg.com/doc/view/1.2.198.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdDrawIndexed-None-02697)
 ERROR 21_descriptor_layout > (VALIDATION) Validation Error: [ UNASSIGNED-CoreValidation-DrawState-DescriptorSetNotBound ] Object 0: handle = 0x2bf5b2c3010, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0xcde11083 | vkCmdDrawIndexed(): VkPipeline 0xe7e6d0000000000f[] uses set #0 but that set is not bound.
 ERROR 21_descriptor_layout > (VALIDATION) Validation Error: [ VUID-vkCmdDrawIndexed-None-02697 ] Object 0: handle = 0xe7e6d0000000000f, type = VK_OBJECT_TYPE_PIPELINE; Object 1: handle = 0x967dd1000000000e, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; Object 2: VK_NULL_HANDLE, type = VK_OBJECT_TYPE_PIPELINE_LAYOUT; | MessageID = 0x9888fef3 | vkCmdDrawIndexed(): VkPipeline 0xe7e6d0000000000f[] defined with VkPipelineLayout 0x967dd1000000000e[] is not compatible for maximum set statically used 0 with bound descriptor sets, last bound with VkPipelineLayout 0x0[] The Vulkan spec states: For each set n that is statically used by the VkPipeline bound to the pipeline bind point used by this command, a descriptor set must have been bound to n at the same pipeline bind point, with a VkPipelineLayout that is compatible for set n, with the VkPipelineLayout used to create the current VkPipeline, as described in Pipeline Layout Compatibility (https://vulkan.lunarg.com/doc/view/1.2.198.1/windows/1.2-extensions/vkspec.html#VUID-vkCmdDrawIndexed-None-02697)
 ERROR 21_descriptor_layout > (VALIDATION) Validation Error: [ UNASSIGNED-CoreValidation-DrawState-DescriptorSetNotBound ] Object 0: handle = 0x2bf5b2c4650, type = VK_OBJECT_TYPE_COMMAND_BUFFER; | MessageID = 0xcde11083 | vkCmdDrawIndexed(): VkPipeline 0xe7e6d0000000000f[] uses set #0 but that set is not bound.
```

Now this might be fully my fault (maybe I installed Vulkan wrong). But I am getting them in my own project as well as in your version on Github.

I ran this to verify that I followed the tutorials well:

```sh
git clone https://github.com/KyleMayes/vulkanalia.git
cd ./vulkanalia/tutorial
cargo run --bin 21_descriptor_layout
```

The same errors are happening there.

I wanted to let you know in case your tutorial is outdated or broken so you can fix it.

Anyways, have a nice day!

### System information

Vulkan version: 1.2.198.1
GPU: NVIDIA GeForce GTX 1060 6GB

Dependency versions

```toml
[dependencies]
anyhow = "1"
lazy_static = "1"
log = "0.4"
nalgebra-glm = "0.10"
png = "0.16"
pretty_env_logger = "0.4"
thiserror = "1"
tobj = "2"
vulkanalia = { version = "=0.14.0", features = ["libloading", "window"] }
winit = "0.24"
```

**Comment by KyleMayes** [comment]
The code for this chapter is expected to raise those errors.
The following chapter adds the descriptor sets the errors are complaining about not being bound.

However, this is not clear and I was confused for a bit about what was going on.
I eventually remembered that this was intentional after running the C++ code for the corresponding chapter from the original tutorial this one is based on and seeing that it also raised the same errors.

I've added a note to the end of this chapter that indicates that these errors are expected.
Thank you for raising this issue!
